src/components/POLine/POLineForm.js

```diff
@@ -240,7 +239,8 @@ class POLineForm extends Component {
     const identifierTypes = getIdentifierTypesForSelect(parentResources);
     const contributorNameTypes = getContributorNameTypesForSelect(parentResources);
     const orderTemplates = getOrderTemplatesForSelect(parentResources);
-    const locations = getLocationOptions(parentResources?.locations?.records || []);
+    const locations = parentResources?.locations?.records ?? [];
```

Comment:
Don't use empty arrays or objects as default values, since it hides the state of data - component to whom you pass such things can't know if it was loaded and empty or it will change later. 
Particularly in this case IIRC you're using didmount effect in field location, so it may be a real problem.